droidlet/perception/semantic_parsing/nsp_transformer_model/optimizer_warmup.py

```diff
@@ -39,9 +49,15 @@ def __init__(self, model, args):
         self._step = 0
 
     def _update_rate(self, stack):
-        return self.lr[stack] * min(
-            (self._step / self.warmup_steps[stack]), (self._step / self.warmup_steps[stack]) ** 0.5
-        )
+        if self._step < self.warmup_steps[stack]:
```

Comment:
is this copied from somewhere? can you cite?

Reply:
This is implemented from [multi_step_lr_scheduler](https://github.com/facebookresearch/mmf/blob/582c7195cbf1eb948436b66c1e9e4bb2e5652a27/mmf/modules/schedulers.py) in [mmf](https://github.com/facebookresearch/mmf). But much simpler. I will improve it to a complete version.